### PR TITLE
[APPAI-1414] Resolve vulnerabilities for dependencies in version comparator utility

### PR DESCRIPTION
**Pull request by dgpatelgit**
Resolve vulnerabilities for dependencies in version comparator utility

Jira issue :: https://issues.redhat.com/browse/APPAI-1414

Checklist ::
- [X] Move any test specific packages from requirements.in to tests/requirements.in
- [ ] (N/A) Move any direct pip installations from Dockerfile.* to requirements.in
- [X] Move any direct pip installations from tests/*.sh to tests/requirements.in
- [X] Relative versioning in requirements.in & tests/requirements.in (no version pinning unless otherwise needed)
- [X] Generate requirements.txt and tests/requirements.txt
- [X] Update setup.py to use requirement.in

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1,2 @@
+typing==3.7.4.3
+codecov==2.1.8
```
Comment (arajkumar) [inline requirements.in]: We shouldn't pin version in requirements.in unless otherwise needed. It can have criteria for versions.

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: requirements.in (hunk 1 above)
Comment (arajkumar) [reply requirements.in]: BTW, codecov is a test dependency. It shouldn't be part of main requirements.in. I would suggest to create a separate test-requirements.{in,txt} & use it from cico scripts.

File: qa/runtests.sh [hunk 2]
```diff
@@ -31,17 +31,15 @@ function prepare_venv() {
...
-
+    python3 "$(which pip3)" install -r tests/requirements.txt
 }
 
 check_python_version
 
 [ "$NOVENV" == "1" ] || prepare_venv || exit 1
 
-$(which pip3) install pytest
 $(which pip3) install pytest-cov
```
Comment (arajkumar) [inline qa/runtests.sh]: Though it is not relevant to this PR, can you move pytest-cov to `tests/requirements.txt`?

File: requirements.in [hunk 3]
```diff
@@ -0,0 +1 @@
+typing
```
Comment (arajkumar) [inline requirements.in]: Since we already on python 3.6, we may not need typing.

**Review by arajkumar — CHANGES_REQUESTED** [review]

**Review by prashbnair — APPROVED** [review]

**Review by arajkumar — CHANGES_REQUESTED** [review]
@dgpatelgit Can you remove the unused requirements.txt from the root directory, of course the same has to be removed from setup.py.

**Comment by dgpatelgit** [comment]
> @dgpatelgit Can you remove the unused requirements.txt from the root directory, of course the same has to be removed from setup.py.

This is done.

**Review by arajkumar — APPROVED** [review]
lgtm

**Comment by dgpatelgit** [comment]
> lgtm

I will verify this changes with other component and then merge.

**Comment by arajkumar** [comment]
All changes are related to tests, so most probably it won't cause any issues. However it is good to test it :)

File: setup.py [hunk 4]
```diff
@@ -5,23 +5,8 @@
 from setuptools import setup, find_packages
 
 
-def get_requirements():
```
Comment (yzainee-zz) [inline setup.py:8]: any specific reasons for removing this code?

Comment (yzainee-zz) [reply setup.py:8]: @dgpatelgit ^^

Comment (dgpatelgit) [reply setup.py:8]: As there is no specific dependencies for this component, we don't need function. There is no requirements.in file for this component. So cleaned up the code.

**Pull request merged**